File: app/models/rol.py
from sqlalchemy import ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Rol(Base):
    __tablename__ = "rol"

    id_rol: Mapped[int] = mapped_column(primary_key=True)
    id_tipo: Mapped[int] = mapped_column(ForeignKey("tipo_rol.id_tipo"))
    tipo: Mapped[str] = mapped_column(unique=True)

File: flutter_app/lib/models/rol.dart
class Rol {
  final int idRol;
  final int idTipo;
  final String tipo;

  Rol({required this.idRol, required this.idTipo, required this.tipo});

  factory Rol.fromJson(Map<String, dynamic> json) {
    return Rol(
      idRol: json['id_rol'] as int,
      idTipo: json['id_tipo'] as int,
      tipo: json['tipo'] as String,
    );
  }
}
